Always use direct vertical aiming when free look is enabled

ceski-1 opened the pull request:
Free look is allowed for single player, multiplayer, and demo recording. That hasn't changed. Direct vertical aiming has historically been a confusing option and overlooked by users who wonder why they can look up/down but can't aim up/down. This PR removes the "Direct Vertical Aiming" menu option and automatically enables it in single player when free look is enabled. Direct vertical aiming is still disabled for multiplayer and demo recording.

rfomin commented:
I often use freelook, but I never use direct aiming. The game wasn't designed with it in mind; from a gameplay perspective, it feels broken to me. Maybe I'm too conservative or hardcore, but I need that option.

It might also break some levels. For example, the player might shoot somewhere they're not supposed to and open a door too early, which could lead to a soft lock.

MrAlaux commented:
If I had to guess, freelook + direct vertical aiming is a much more common combo than freelook + autoaim; we could enable d.v.a. by default to avoid the confusion while still allowing users like rfomin to have only freelook.

ceski-1 commented:
Here's another approach. Free look is allowed for single player and disabled for both multiplayer and demo recording. It now has three options: "Off", "Auto-Aim", and "Direct Aim". Mouse and gamepad free look share the same config setting: `freelook`. Both "Free Look" menu items are retained in the "Mouse" and "Gamepad" tabs. Removed "Direct Vertical Aiming" menu option. Removed `mouselook` and `padlook` config settings.

There's one issue though. The "Free Look" key binding doesn't work conceptually when there are three options. Cycling through all three is annoying, so that behavior is not used. Toggling between "Off" and the last preferred option doesn't work right in practice. That is, toggling the key, changing the menu item, and relaunching the game will eventually lead to an undesirable state where the behavior doesn't match expectations. It's hard to put into words. I'm not sure how to solve this yet.

Maybe it's best to close this PR and try Alaux's idea as a compromise.

rfomin commented:
Another idea:
```
     Freelook: On/Off
Freelook Mode: Auto-Aim/Direct
      ...
```

ceski-1 commented:
> Another idea:

Unfortunately, there's no more room in the Gamepad menu.

rfomin commented:
> One example:

We need two configuration variables: "on/off" and "mode." In the General menu, we switch the "mode", the "on/off" variable will be toggled automatically in the `.action` function. The freelook key will toggle only the "on/off" variable.

Pedro-Beirao commented:
> Free look is allowed for single player, multiplayer, and demo recording

Is it allowed in strict demo recording? That isnt allowed by dsda rules.

rfomin commented:
> > Free look is allowed for single player, multiplayer, and demo recording
> 
> Is it allowed in strict demo recording? That isnt allowed by dsda rules.

"Direct Aim" mode is not allowed because it causes desync. "Auto Aim" mode is allowed. Thanks, we will disable it, too.

ceski-1 commented:
Closing in favor of https://github.com/fabiangreffrath/woof/pull/2332.
